File: useUIView/useUIView/ViewController.m
//
//  ViewController.m
//  useUIView
//
//  Created by admin on 16/5/27.
//  Copyright © 2016年 yicheng. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@property (weak, nonatomic) IBOutlet UIView *yellowView;
@property (weak, nonatomic) IBOutlet UIView *blackView;
@property (weak, nonatomic) IBOutlet UIView *redView;

@property (weak, nonatomic) IBOutlet UIView *greenView;
@property (weak, nonatomic) IBOutlet UIView *blueView;


@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    [self printPosition:_yellowView viewName:@"yellowView"];
    [self printPosition:_blackView viewName:@"blackView"];
    [self printPosition:_redView viewName:@"redView"];
    
    [self printPosition:_greenView viewName:@"greenView"];
    [self printPosition:_blueView viewName:@"blueView"];
    
}

/** 打印frame, bounds, center 信息. */
-(void)printPosition:(UIView *)view viewName:(NSString *)name {
    NSLog(@"%@ frame = %@", name, NSStringFromCGRect(view.frame));
    NSLog(@"%@ bounds = %@", name, NSStringFromCGRect(view.bounds));
    NSLog(@"%@ center = %@", name, NSStringFromCGPoint(view.center));
}

@end

File: Transform/Transform/ViewController.swift
//
//  ViewController.swift
//  Transform
//
//  Created by admin on 16/6/28.
//  Copyright © 2016年 yicheng. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    @IBOutlet weak var redView: UIView!
    @IBOutlet weak var blueView: UIView!
    @IBOutlet weak var greenView: UIView!
    @IBOutlet weak var blackView: UIView!
    
    var isRunning: Bool = true
    var timer: NSTimer?
    
    override func viewDidLoad() {
        super.viewDidLoad()

    }
    
    override func touchesBegan(touches: Set<UITouch>, withEvent event: UIEvent?) {

        if isRunning {
            timer = NSTimer.scheduledTimerWithTimeInterval(1,
                                                               target: self,
                                                               selector: #selector(ViewController.action),
                                                               userInfo: nil,
                                                               repeats: true)
        } else {
            timer?.invalidate()
        }
        isRunning = !isRunning
    }
    
    /** 让4个 view 都动起来. */
    func action() {
        UIView.animateWithDuration(1,
                                   delay: 0,
                                   options: UIViewAnimationOptions.Repeat,
                                   animations: {
                                    self.randomAnimation(self.redView)
                                    self.randomAnimation(self.blueView)
                                    self.randomAnimation(self.greenView)
                                    self.randomAnimation(self.blackView)
            }, completion: nil)
    }
    
    /** 随机动画. */
    func randomAnimation(myView: UIView) {
        let flag = Int(arc4random_uniform(uint(4)))
        
        var myUIViewAnimationTransition: UIViewAnimationTransition;
        switch flag {
        case 1:
            myUIViewAnimationTransition = .FlipFromLeft
        case 2:
            myUIViewAnimationTransition = .FlipFromRight
        case 3:
            myUIViewAnimationTransition = .CurlDown
        case 4:
            myUIViewAnimationTransition = .CurlUp
        default:
            myUIViewAnimationTransition = .FlipFromLeft
        }
        
        UIView.beginAnimations(nil, context: nil)
        UIView.setAnimationTransition(myUIViewAnimationTransition, forView: myView, cache: true)
        UIView.setAnimationDuration(1.0)
        UIView.setAnimationRepeatAutoreverses(true)
        UIView.commitAnimations()
    }
}

